haystack/preview/components/retrievers/memory.py

```diff
@@ -39,28 +40,54 @@ class Output(ComponentOutput):
 
         documents: List[Document]
 
-    def __init__(
-        self,
-        document_store_name: str,
-        filters: Optional[Dict[str, Any]] = None,
-        top_k: int = 10,
-        scale_score: bool = True,
-    ):
+    def __init__(self, filters: Optional[Dict[str, Any]] = None, top_k: int = 10, scale_score: bool = True):
         """
         Create a MemoryRetriever component.
 
-        :param document_store_name: The name of the MemoryDocumentStore to retrieve documents from.
         :param filters: A dictionary with filters to narrow down the search space (default is None).
         :param top_k: The maximum number of documents to retrieve (default is 10).
         :param scale_score: Whether to scale the BM25 score or not (default is True).
 
         :raises ValueError: If the specified top_k is not > 0.
         """
-        self.document_store_name = document_store_name
         if top_k <= 0:
             raise ValueError(f"top_k must be > 0, but got {top_k}")
         self.defaults = {"top_k": top_k, "scale_score": scale_score, "filters": filters or {}}
 
+    @property
+    def store(self) -> Optional[MemoryDocumentStore]:
+        """
+        This property allows Pipelines to connect the component with the stores it requires.
+
+        Stores have to be instances of MemoryDocumentStore, or the assignment will fail.
+        """
+        return getattr(self, "_store", None)
+
+    @store.setter
+    def store(self, store: MemoryDocumentStore):
+        """
+        This property allows Pipelines to connect the component with the stores it requires.
+
+        Stores have to be instances of MemoryDocumentStore, or the assignment will fail.
+
+        :param store: the MemoryDocumentStore instance to retrieve from.
+        :raises ValueError if the store is not an instance of MemoryDocumentStore.
+        """
+        if not store:
+            raise ValueError("Can't set the value of the store to None.")
+        if not isinstance(store, MemoryDocumentStore):
+            raise ValueError("MemoryRetriever can only be used with a MemoryDocumentStore instance.")
+        self._store = store
+
+    def warmup(self):
+        """
+        Double-checks that a store is set before running this component in a pipeline.
+        """
+        if not self.store:
+            raise ValueError(
+                "MemoryRetriever needs a store to run: use the 'store' parameter of 'add_component' to connect them."
+            )
```

Comment:
Isn't this sort of validation?

Reply:
Yes and I hate that I had to put it in here... where shall I put it? In `run()` is far too late imho, but in `__init__` it makes impossible to initialize it.

Reply:
I mean, I also have it in run(), but that's only for direct calls to the method, outside of Pipelines. Pipelines should have a way to do this kind of checks earlier and warm_up seems like the only valid place. Open to suggestions